miio/tests/test_ceil.py

```diff
@@ -0,0 +1,126 @@
+from unittest import TestCase
+from miio import Ceil
+from .dummies import DummyDevice
+import pytest
+
+
+class DummyCeil(DummyDevice, Ceil):
+    def __init__(self, *args, **kwargs):
+        self.state = {'power': 'on',
+                      'bright': 50,
+                      'snm': 4,
+                      'dv': 0,
+                      'cctsw': [[0, 3], [0, 2], [0, 1]],
+                      'bl': 1,
+                      'mb': 1,
+                      'ac': 1,
+                      'mssw': 1,
+                      'cct': 99
+                      }
+        self.return_values = {
+            'get_prop': self._get_state,
+            'set_power': lambda x: self._set_state("power", x),
+            'set_bright': lambda x: self._set_state("bright", x),
+            'apply_fixed_scene': lambda x: self._set_state("snm", x),
+            'delay_off': lambda x: self._set_state("dv", x),
+            'enable_bl': lambda x: self._set_state("bl", x),
+            'enable_ac': lambda x: self._set_state("ac", x),
+            'set_cct': lambda x: self._set_state("cct", x),
+        }
+        super().__init__(args, kwargs)
+
+
+@pytest.fixture(scope="class")
+def ceil(request):
+    request.cls.device = DummyCeil()
+    # TODO add ability to test on a real device
+
+
+@pytest.mark.usefixtures("ceil")
+class TestCeil(TestCase):
+    def is_on(self):
+        return self.device.status().is_on
+
+    def state(self):
+        return self.device.status()
+
+    def test_on(self):
+        self.device.off()  # ensure off
+
+        start_state = self.is_on()
+        assert start_state is False
+
+        self.device.on()
+        assert self.is_on() is True
+
+    def test_off(self):
+        self.device.on()  # ensure on
+
+        assert self.is_on() is True
+        self.device.off()
+        assert self.is_on() is False
```

Comment:
Considering quite a few devices has the ability for turning on and off, maybe it makes sense to have a single set of tests for all of those devices. For now this is fine though.